### PR TITLE
UPPSF-4974-Notification-for-RELATEDCONTENT-only-if-policy-INTERNAL_UNSTABLE

### PR DESCRIPTION
**Pull request by tsvetko24**
# Description

Change aims to send new type notification RELATEDCONTENT only for subscribes of policy INTERNAL_UNSTABLE.

## What

Added filed to save the Policy of the Subscribers and check this field and notification type when decide for notification send.
Could be implemented with a new Policy as well. 

## Why

https://financialtimes.atlassian.net/browse/UPPSF-4974 

## Anything, in particular, you'd like to highlight to reviewers

Mention here sections of code which you would like reviewers to pay extra attention to .E.g

_Would appreciate a second pair of eyes on the test_  
_I am not quite sure how this bit works_  

## Scope and particulars of this PR (Please tick all that apply)

- [x] Tech hygiene (dependency updating & other tech debt)
- [ ] Bug fix
- [x] Feature
- [x] Documentation
- [ ] Breaking change
- [ ] Minor change (e.g. fixing a typo, adding config)

## DoD - Ensure all relevant tasks are completed before marking this PR as "Ready for review"

- [x] Test coverage is not significantly decreased
- [x] All PR checks have passed
- [x] Changes are deployed on dev before asking for review
- [ ] Documentations remains up-to-date
  - [ ] OpenAPI definition file is updated
  - [ ] README file is updated
  - [ ] Documentation is updated in upp-docs and upp-public-docs
  - [ ] Architecture diagrams are updated

___
This Pull Request follows the rules described in our [Pull Requests Guide](https://github.com/Financial-Times/upp-docs/tree/master/guides/pr-guide)

### CODE REVIEW

File: dispatch/dispatcher.go [hunk 1]
```diff
@@ -214,6 +215,11 @@ func (d *Dispatcher) forwardToSubscribers(notification NotificationModel) {
 				entry.Info("Skipping subscriber due to blocked publication.")
 				continue
 			}
+			if isRelatedContent && !sub.Options().ReceiveInternalUnstable {
```
Comment (vlasakievft) [inline dispatch/dispatcher.go:218]: **suggestion**:
```go
if strings.Contains(notifications.Type, ContentRelationType) {
   //
}
```

Comment (tsvetko24) [reply dispatch/dispatcher.go:218]: Logic has been changed, so we do not have ContentRelationType anymore, and we check the Payload, not the type. Hope this is resolved with the change.

**Review by vlasakievft — DISMISSED** [review]
🚀

**Review by todor-videv1 — APPROVED** [review]
LGTM

**Review by antoanFT — APPROVED** [review]

**Pull request merged**